consensus/spos/worker.go

```diff
@@ -783,22 +784,20 @@ func (wrk *Worker) processEquivalentMessageUnprotected(cnsMsg *consensus.Message
 		return err
 	}
 
-	// TODO[Sorin next PR]: update EquivalentMessageInfo structure to hold also the proof(bitmap+signature) that was received
-	// then on commit block store this data on blockchain in order to use it on the next block creation
 	equivalentMsgInfo.Validated = true
+	equivalentMsgInfo.PreviousPubkeysBitmap = cnsMsg.PubKeysBitmap
+	equivalentMsgInfo.PreviousAggregateSignature = cnsMsg.AggregateSignature
 
 	return nil
 }
 
-func (wrk *Worker) verifyEquivalentMessageSignature(_ *consensus.Message) error {
+func (wrk *Worker) verifyEquivalentMessageSignature(cnsMsg *consensus.Message) error {
 	if check.IfNil(wrk.consensusState.Header) {
 		return ErrNilHeader
 	}
 
-	header := wrk.consensusState.Header.ShallowClone()
-
-	// TODO[Sorin]: after flag enabled, VerifySignature on previous hash, with the signature and bitmap from the proof on cnsMsg
-	return wrk.headerSigVerifier.VerifySignature(header)
+	header := wrk.consensusState.Header
+	return wrk.headerSigVerifier.VerifySignatureForHash(header, header.GetPrevHash(), cnsMsg.PubKeysBitmap, cnsMsg.Signature)
```

Comment:
the verification for the equivalent proof should be done on the current header hash not on the previous one.